src/qt/bitcoingui.cpp

```diff
@@ -707,6 +714,24 @@ void BitcoinGUI::openClicked()
     }
 }
 
+void BitcoinGUI::openWalletClicked()
+{
+    QString fileName = QFileDialog::getExistingDirectory(this, tr("Open Wallet"),
+                                                QString(),
```

Comment:
Suggestion: start in the data directory for the current network (mainnet/testnet/regtest) instead of the home directory, since that's where the wallet files and folders will usually be (and because it may be difficult to traverse down into `.bitcoin`).